packages/check/match.js

```diff
@@ -192,6 +192,17 @@ const stringForErrorMessage = (value, options = {}) => {
   return EJSON.stringify(value);
 };
 
+/**
+ * @summary Returns true if the value is a non empty string.
+ * @locus Anywhere
+ * @param {Any} value The value to check
+ */
+
+export const NonEmptyString = Match.Where((x) => {
+  check(x, String);
+  return x.length > 0;
+});
```

Comment:
I'd like to hear what others think, but I don't see why it's a separate export and not a `Match` property, just like `Maybe` or `Where`.

Reply:
Good point, but then I think it would be better to transform this into general `NonEmpty` function where you could put in type:
```
check(userId, Match.NonEmpty(String))
```

Reply:
Good idea! But it should have a list of allowed types, since `Match.NonEmpty(CustomType)` would require additional implementation in the `CustomType`.